src/components/SaveEventButton.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React from "react";
+import { Animated, Easing, StyleSheet } from "react-native";
+import LottieView from "lottie-react-native";
+import Touchable from "./Touchable";
+import heartAnimationLight from "../../assets/animations/save-event-light.json";
+import heartAnimationDark from "../../assets/animations/save-event-dark.json";
+
+type Props = {
+  active: boolean,
+  onDark: boolean,
+  onPress: boolean => void
+};
+
+type State = {
+  progress: Object
+};
+
+export default class SaveEventButton extends React.Component<Props, State> {
+  static defaultProps = {
+    active: false,
+    onDark: false
+  };
+
+  state = {
+    progress: new Animated.Value(0)
+  };
+
+  componentDidMount() {
+    if (this.props.active) {
+      this.state.progress = new Animated.Value(1);
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const value = this.props.active ? 1 : 0;
+    if (this.props.active === true && prevProps.active === false) {
+      Animated.timing(this.state.progress, {
+        toValue: value,
+        duration: 1920,
+        easing: Easing.linear,
+        useNativeDriver: true
+      }).start();
+      return;
+    }
+    Animated.timing(this.state.progress, {
+      toValue: value,
+      duration: 0,
+      easing: Easing.linear,
+      useNativeDriver: true
+    }).start();
```

Comment:
Why is the animation with duration `0` needed? The start value for `state.progress` should always be correct because of `componentDidMount`. So all changes to `active` should be covered already. And when another prop than `active` changes while an animation is in progress, this will most likely interrupt the animation.

Reply:
This is to toggle the button to a value state immediately. For example when the component mounts we do not want the animation to play. This should only happen when the button is not active and changes to active.

Reply:
But isn't this what `componentDidMount` is for?

Reply:
I have refactored, but this still needs to be present for the toggling active from true to false case.